mktrayicon.c

```diff
@@ -237,9 +238,13 @@ int main(int argc, char **argv)
 		gtk_status_icon_set_tooltip_text(icon, argv[4]);
 	}
 
-	if (/* test to see if last arg is a pipe */) {
+	/* test if last argument is a pipe */
+	char *command = malloc(1024*sizeof(char));
+	sprintf(command, "test -p %s", argv[argc-1]);
```

Comment:
I don't think you need this? If the last argument is a `-t`, or the last argument followed a `-i` or `-t`, then we know it isn't a pipe.

Reply:
It's an eyesore, I know. But I can't think of anything else that would be foolproof. We could test if next-to-last argument contained a `-` but that would fail if call was `mktrayicon <FIFO>`

Reply:
Here's what I would do:
 - Set `args = 1`
 - If `-i`, do `args += 2`
 - If `-t`, do `args += 2`
 - If `args < argc`, `argv[argc - 1]` is a FIFO

Reply:
Cool. Done. 

For some reason it still segfaults on `mktrayicon -i <ICON> &`. Any ideas?

Reply:
Oh, I think argv[3] becomes a problem if there are only two arguments.